### PR TITLE
[S.R.I.RuntimeInformation] Fix FrameworkDescription on Android

### PR DESCRIPTION
**Pull request by akoeplinger**
It turns out using `DllImport ["__Internal"]` doesn't work on XA (and other cases where Mono is embedded) because it searches for the method in the host which embeds Mono and doesn't find it, resulting in an EntryPointNotFoundException.

We need to switch to an icall instead. However there's already an existing icall+BCL wrapper: `Mono.Runtime.GetDisplayName()`.

Added IVT from corlib to S.R.I.RuntimeInformation so we can call it.

### CODE REVIEW

File: mcs/class/Facades/System.Runtime.InteropServices.RuntimeInformation/System.Runtime.InteropServices/RuntimeInformation.cs [hunk 1]
```diff
@@ -35,12 +35,14 @@ namespace System.Runtime.InteropServices
...
 	{
-		[DllImport ("__Internal")]
-		extern static string mono_get_runtime_build_info ();
+		static string frameworkDescription;
 
 		public static string FrameworkDescription {
 			get {
-				return mono_get_runtime_build_info ();
+				if (frameworkDescription == null)
+					frameworkDescription = "Mono " + (string)Type.GetType ("Mono.Runtime").GetMethod ("GetDisplayName", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic).Invoke (null, null);
```
Comment (marek-safar) [inline mcs/class/Facades/System.Runtime.InteropServices.RuntimeInformation/System.Runtime.InteropServices/RuntimeInformation.cs]: This is not good solution as it breaks linker for example. Better to do InternalsVisibleTo from mscorlib or add new icall

Comment (akoeplinger) [inline mcs/class/Facades/System.Runtime.InteropServices.RuntimeInformation/System.Runtime.InteropServices/RuntimeInformation.cs]: `Mono.Runtime` is already a special type that the linker knows about and preserves as it's often used via reflection only: https://github.com/mono/mono/blob/master/mcs/class/corlib/LinkerDescriptor/mscorlib.xml#L839-L845

Comment (akoeplinger) [inline mcs/class/Facades/System.Runtime.InteropServices.RuntimeInformation/System.Runtime.InteropServices/RuntimeInformation.cs]: I can't add another icall as that only works from mscorlib.dll. I'll try IVT :)

**Comment by akoeplinger** [comment]
Changed to the IVT solution.

**Comment by kumpera** [comment]
Did you follow up with the XA team to fix __Internal? This is a bug on their end.

**Comment by akoeplinger** [comment]
@kumpera no, I thought the behavior was by design. I'll talk to them.

**Pull request merged**